lib/dal/src/workspace_snapshot.rs

```diff
@@ -355,18 +355,22 @@ impl WorkspaceSnapshot {
         let self_clone = self.clone();
         let updates = slow_rt::spawn(async move {
             let mut working_copy = self_clone.working_copy_mut().await;
-            working_copy.cleanup();
+            working_copy.cleanup_and_merkle_tree_hash()?;
 
-            self_clone.read_only_graph.detect_updates(&working_copy)
+            Ok::<Vec<Update>, WorkspaceSnapshotGraphError>(
+                self_clone.read_only_graph.detect_updates(&working_copy),
+            )
         })?
-        .await?;
+        .await??;
```

Comment:
Took me a few times to read but got it. You're directly returning the updates because now the cleanup method is fallible and you need to be explicit about the return type.

Reply:
yeah the async closure needs to get a return type somewhere so that `?` works